src/main/java/Bosco.java

```diff
@@ -0,0 +1,96 @@
+import java.sql.SQLOutput;
+import java.util.Scanner;
+
+public class Bosco {
+    private static final String DIVIDER = "\t____________________________________________________________";
+    private static final int MAX_TASKS = 100;
+
+    private static final Task[] tasks = new Task[MAX_TASKS];
+    private static int taskCount = 0;
+
+    private static void printWelcomeMessage() {
+        System.out.println(DIVIDER);
+        System.out.println("\t Hello! I'm Bosco APD." + System.lineSeparator() + "\t What can I do for you?");
+        System.out.println(DIVIDER);
+    }
+
+    private static void printExitMessage() {
+        System.out.println(DIVIDER);
+        System.out.println("\t Bye! Hope to see you again soon!");
+        System.out.println(DIVIDER);
+    }
+
+    public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
+        String userInputString;
+        String[] words;
+        Task selectedTask;
+
+        printWelcomeMessage();
+
+        while (true) {
+            userInputString = in.nextLine();
+            words = userInputString.split("\\s+", 2);
+            String commandType = words[0];
+            if (commandType.equals("bye")) {
+                break;
+            }
+            String commandArgs = (words.length == 2 ? words[1] : words[0]);
+
+            System.out.println(DIVIDER);
+            switch (commandType) {
+            case "list":
+                for (int i = 0; i < taskCount; i++) {
+                    System.out.println("\t " + (i + 1) + "." + tasks[i]);
+                }
```

Comment:
Perhaps the lines of code for the separate cases can be abstracted out? For example as a method printList(), markTask(), addTodo() etc.

Reply:
Agreed, I will be adding more abstraction in future improvements.